queue.go

```diff
@@ -260,6 +269,40 @@ func (q *DQue) Peek() (interface{}, error) {
 	return obj, nil
 }
 
+// DequeueBlock behaves similar to Dequeue, but is a blocking call until an item is available.
+func (q *DQue) DequeueBlock() (interface{}, error) {
+	for {
+		obj, err := q.Dequeue()
+		if err == ErrEmpty {
+			q.mutexEmptyCond.Lock()
+			q.emptyCond.Wait()
+			q.mutexEmptyCond.Unlock()
```

Comment:
It feels like there is room for a race condition here...  where you could get an ErrEmpty, but something is placed into the queue before you hit the Wait().  So then you'd be waiting when there was something in the queue.

Reply:
You are right - I pushed a fix (more like a rewrite).